src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -526,192 +461,171 @@ protected Future<PoLineAndTitleById> getPoLineAndTitleById(List<String> poLineId
+      for (Piece piece : pieceList) {
+        if (StringUtils.isNotEmpty(piece.getItemId())) {
+          itemIdsByTenantId.computeIfAbsent(piece.getReceivingTenantId(), k -> new ArrayList<>()).add(piece.getItemId());
+        }
+      }
     }
 
-    ReceivingItemResult itemResult = new ReceivingItemResult();
-    itemResult.setPieceId(pieceId);
-    itemResult.setProcessingStatus(status);
-    result.getReceivingItemResults().add(itemResult);
+    // split all id lists by maximum number of id's for get query
+    List<Future<List<JsonObject>>> futures = itemIdsByTenantId.entrySet()
+      .stream()
+      .flatMap(
+        entry -> StreamEx.ofSubLists(entry.getValue(), MAX_IDS_FOR_GET_RQ_15)
+          .map(ids -> {
+            var locationContext = RequestContextUtil.createContextWithNewTenantId(requestContext, entry.getKey());
+            return getItemRecordsByIds(ids, piecesByItemId, locationContext);
+          })
+      )
+      .toList();
+
+    return collectResultsOnSuccess(futures)
+      .map(lists -> StreamEx.of(lists).toFlatList(jsonObjects -> jsonObjects));
   }
 
   /**
-   * Filter by locationId presence for items/pieces related to POLine.
+   * Returns list of item records for specified id's.
    *
-   * @return {@link Future} which holds map with PO line id as key
-   *         and list of corresponding pieces as value
+   * @param ids             List of item id's
+   * @param piecesWithItems map with item id as a key and piece record as a value
+   * @return future with list of item records
    */
-  Future<Map<String, List<Piece>>> filterMissingLocations(Map<String, List<Piece>> piecesRecords, RequestContext requestContext) {
-    return getPoLines(StreamEx.ofKeys(piecesRecords).toList(), requestContext)
-      .map(poLines -> {
-        for(PoLine poLine : poLines) {
-          piecesRecords.get(poLine.getId()).removeIf(piece -> isMissingLocation(poLine, piece));
-        }
-        return piecesRecords;
+  private Future<List<JsonObject>> getItemRecordsByIds(List<String> ids, Map<String, Piece> piecesWithItems, RequestContext requestContext) {
+    return inventoryItemManager.getItemRecordsByIds(ids, requestContext)
+      .map(items -> {
+        checkIfAllItemsFound(ids, items, piecesWithItems);
+        return items;
+      })
+      .otherwise(e -> {
+        logger.error("The issue happened getting item records");
+        ids.forEach(id -> {
+          Piece piece = piecesWithItems.get(id);
+          addError(piece.getPoLineId(), piece.getId(), ITEM_NOT_RETRIEVED.toError());
+        });
+        return Collections.emptyList();
       });
   }
 
-  private boolean isMissingLocation(PoLine poLine, Piece piece) {
-    // Check if locationId doesn't presented in piece from request and retrieved from storage
-    // Corresponding piece from collection
-    if ((getLocationId(piece) == null && getHoldingId(piece) == null) && !isRevertToOnOrder(piece)) {
```

Comment:
rewritten version is: 
(getLocationId(piece) != null || getHoldingId(piece) != null || isRevertToOnOrder(piece))
so previous version is not the same because it checks that both holdingId and locationId are null, and rewritten version checks one by one.

Reply:
old version:
```
if (a && b && c) {
    // some code
}
return false;
```
this is equivalent to:
```
if (!a || !b || !c) {
    return false;
}
// some code
```